cmd/oras/blob/fetch.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package blob
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/oci"
+
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/internal/cache"
+	"oras.land/oras/internal/cas"
+)
+
+type fetchBlobOptions struct {
+	option.Common
+	option.Descriptor
+	option.Pretty
+	option.Remote
+
+	cacheRoot  string
+	outputPath string
+	targetRef  string
+}
+
+func fetchCmd() *cobra.Command {
+	var opts fetchBlobOptions
+	cmd := &cobra.Command{
+		Use:   "fetch <name@digest> [flags]",
+		Short: "[Preview] Fetch a blob from a remote registry",
+		Long: `[Preview] Fetch a blob from a remote registry
+** This command is in preview and under development. **
+
+Example - Fetch raw blob:
+  oras blob fetch localhost:5000/hello@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5
+
+Example - Fetch the descriptor of a blob:
+  oras blob fetch --descriptor localhost:5000/hello@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5
+
+Example - Fetch blob from the insecure registry:
+  oras blob fetch localhost:5000/hello@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5 --insecure
```

Comment:
nit: keep the flag in the same position (before arguments) in all examples.

Reply:
Made requested change.